File: backend/internal/models/daily_reward.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type DailyReward struct {
	UserID        uuid.UUID
	LastClaimedAt time.Time
	StreakDays    int32
}

// Today returns today's date in UTC as a string (YYYY-MM-DD)
func Today() string {
	return time.Now().UTC().Format("2006-01-02")
}

File: backend/src/models/daily_reward.rs
use chrono::{DateTime, Utc};
use uuid::Uuid;

#[derive(Debug, Clone)]
pub struct DailyReward {
    pub user_id: Uuid,
    pub last_claimed_at: DateTime<Utc>,
    pub streak_days: i32,
}

pub fn today_ymd() -> String {
    Utc::now().format("%Y-%m-%d").to_string()
}
